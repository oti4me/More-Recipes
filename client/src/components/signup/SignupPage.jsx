import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SignupForm from './SignupForm';
import Footer from '../Footer';
import Header from '../Header';

/**
 * 
 * 
 * @class SignupPage
 * 
 * @extends {Component}
 */
class SignupPage extends Component {

  /**
   * Creates an instance of SignupPage.
   * 
   * @param {object} props 
   * 
   * @memberof SignupPage
   */
  constructor(props) {
    super(props);
  }

  /**
   * A method that renders the jsx component to the browser
   * 
   * @returns {object} Sign up jsx object
   * 
   * @memberof SignupPage
   */
  render() {
    return (
      <div className="main">
        <Header {...this.props} />
        <div className="container cont">
          <div className="row">
            <div className="col s12  s2 m8 l8 offset-m2 offset-l2">
              <h2 style={{ textAlign: 'center' }}>Sign Up</h2>
              <SignupForm history={this.props.history} />
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

SignupPage.propTypes = {
  history: PropTypes.shape({}).isRequired
}

const mapStateToProps = (state) => {
  return { state };
}

export default connect(mapStateToProps)(SignupPage);
