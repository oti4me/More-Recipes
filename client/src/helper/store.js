import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducers from '../reducers';

// initial state of the app
const initialState = {
  recipes: {},
  auth: {
    user: {},
    loggedIn: false,
  }
};

// create store to save app state
const store = createStore(reducers, initialState,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;