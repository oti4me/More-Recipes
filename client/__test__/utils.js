import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import store from '../src/helper/createStore';

export const mountWrapper = (Component, store, props) => mount(
  <Provider store={store} >
    <BrowserRouter>
      <Component {...props} />
    </BrowserRouter>
  </Provider>
);

export const mainState = store.getState();

export default {
  mountWrapper,
}