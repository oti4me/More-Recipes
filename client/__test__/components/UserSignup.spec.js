import React from 'react';
import toJson from 'enzyme-to-json';
// import ConnectedSignupPage, { SignupPage } from '../../src/components/signup/SignupPage';
import SignupPage from '../../src/components/signup/SignupPage';
// import SignupPage from '../../src/components/signup/SignupPage';
import { mountWrapper } from '../utils';
import mockStore from '../mockData/mockStore';
import store from '../../src/helper/createStore';

const mainState = store.getState();
const props = {
  history: {
    push: () => { },
  }
};

const userInfo = {
  firstName: 'Henry',
  lastName: 'Otighe',
  email: 'otighe@gmail.com',
  phone: '07067143161',
  password: 'Oti4me@gmail.com',
  confirmPassword: 'Oti4me@gmail.com'
}

const newState = {
  ...mainState,
  getFavourites: jest.fn(() => Promise.resolve()),
  auth: {
    user: {
      userId: 3,
    },
    loggedIn: true,
  }
};


const event = (name, value) => ({
  preventDefault: jest.fn(),
  target: {
    name,
    value
  }
});

const newStore = mockStore(newState);
const wrapper = mountWrapper(SignupPage, newStore, props);
const SignupFormWrapper = wrapper.find('SignupForm');

describe('Sign-Up Component Suite', () => {
  it('should render SignupPage correctly without exploding', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should handles onChange action on form fields', () => {
    SignupFormWrapper.instance()
      .handleChange(event('firstName', userInfo.firstName));
    SignupFormWrapper.instance()
      .handleChange(event('lastName', userInfo.lastName));
    SignupFormWrapper.instance()
      .handleChange(event('email', userInfo.email));
    SignupFormWrapper.instance()
      .handleChange(event('phone', userInfo.phone));
    SignupFormWrapper.instance()
      .handleChange(event('password', userInfo.password));
    SignupFormWrapper.instance()
      .handleChange(event('confirmPassword', userInfo.confirmPassword));
    expect(SignupFormWrapper.instance().state)
      .toEqual(userInfo);
  });
});

SignupFormWrapper.find('a').at(0).simulate('click', {
  preventDefault: jest.fn(),
});


// const state = {
//   email: "",
//   password: ""
// };

// const func = jest.fn();
// const props = {
//   loggedIn: true,
//   error: {},
//   history: {
//     push: jest.fn()
//   },
//   signup: jest.fn()
// };
// const signin = jest.fn()

// const errorProps = {
//   loggedIn: true,
//   error: {
//     status: 400,
//     message: [
//       'username cannot be empty'
//     ]
//   },
//   signin: jest.fn()
// };

// const event = {
//   target: {
//     name: 'firstName',
//     value: 'Henry'
//   },
//   preventDefault: jest.fn()
// };

// describe('Signup test suit', () => {

//   it('Should render SignupForm component without exploding', () => {
//     const wrapper = shallow(<SignupForm {...props} />);
//     expect(wrapper).toMatchSnapshot();
//   });

//   it('Should render Signup connected component without exploding', () => {
//     const wrapper = shallow(
//       <ConnectedSignupForm store={store} {...props} />
//     );
//     expect(wrapper.length).toBe(1);
//     expect(wrapper).toMatchSnapshot();
//   });

//   it('Should handle field change', () => {
//     const wrapper = shallow(
//       <SignupForm {...props} {...state} />
//     );
//     const spy = jest.spyOn(wrapper.instance(), 'handleSignup');
//     // wrapper.instance().handleChange(event);
//     wrapper.instance().handleSignup(event);
//     wrapper.setState({
//       firstName: 'Monkey',
//       lastName: 'Homoerectus',
//       email: 'me@gmail.com',
//       password: 'Abcd.123.@',
//       confirmPassword: 'Abcd.123.@',
//       phone: '07033333333'
//     });
//     wrapper.find('#signup').simulate('submit', event);
//     expect(spy).toHaveBeenCalled();
//     expect(wrapper.instance().props.signup).toHaveBeenCalled();
//   });


//   it('Should handle field change', () => {
//     const wrapper = shallow(
//       <SignupForm {...props} {...state} />
//     );
//     const spy = jest.spyOn(wrapper.instance(), 'handleChange');
//     wrapper.instance().handleChange(event);
//     wrapper.find('#fname').simulate('change', event);
//     expect(spy).toHaveBeenCalled();
//     expect(wrapper.instance().state.firstName).toEqual('Henry');
//   });

//   it('Should handle field change', () => {
//     const wrapper = shallow(
//       <ConnectedSignupPage store={store} {...props} {...state} />
//     );
//     expect(wrapper).toMatchSnapshot();
//   });
//   it('Should render SignupPage correctly', () => {
//     const wrapper = shallow(
//       <SignupPage {...props} {...state} />
//     );
//     expect(wrapper).toMatchSnapshot();
//   });
// });
