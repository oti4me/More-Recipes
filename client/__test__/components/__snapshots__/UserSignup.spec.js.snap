// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-Up Component Suite should render SignupPage correctly without exploding 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(SignupPage)
        history={
          Object {
            "push": [Function],
          }
        }
      >
        <SignupPage
          dispatch={[Function]}
          history={
            Object {
              "push": [Function],
            }
          }
          state={
            Object {
              "auth": Object {
                "loggedIn": true,
                "user": Object {
                  "userId": 3,
                },
              },
              "getFavourites": [MockFunction],
              "recipes": Object {},
            }
          }
        >
          <div
            className="main"
          >
            <Connect(Header)
              dispatch={[Function]}
              history={
                Object {
                  "push": [Function],
                }
              }
              state={
                Object {
                  "auth": Object {
                    "loggedIn": true,
                    "user": Object {
                      "userId": 3,
                    },
                  },
                  "getFavourites": [MockFunction],
                  "recipes": Object {},
                }
              }
            >
              <Header
                dispatch={[Function]}
                history={
                  Object {
                    "push": [Function],
                  }
                }
                loggedIn={true}
                signOut={[Function]}
                state={
                  Object {
                    "auth": Object {
                      "loggedIn": true,
                      "user": Object {
                        "userId": 3,
                      },
                    },
                    "getFavourites": [MockFunction],
                    "recipes": Object {},
                  }
                }
                user={
                  Object {
                    "userId": 3,
                  }
                }
              >
                <div>
                  <nav>
                    <div
                      className="nav-wrapper color-green header1"
                    >
                      <Link
                        className="brand-logo"
                        replace={false}
                        style={
                          Object {
                            "fontSize": "22px",
                          }
                        }
                        to="/home"
                      >
                        <a
                          className="brand-logo"
                          href="/home"
                          onClick={[Function]}
                          style={
                            Object {
                              "fontSize": "22px",
                            }
                          }
                        >
                          More-Recipes
                        </a>
                      </Link>
                      <Link
                        className="button-collapse"
                        data-activates="mobile2"
                        replace={false}
                        to="!#"
                      >
                        <a
                          className="button-collapse"
                          data-activates="mobile2"
                          href="!#"
                          onClick={[Function]}
                        >
                          <i
                            className="material-icons"
                          >
                            menu
                          </i>
                        </a>
                      </Link>
                      <ul
                        className="right hide-on-med-and-down"
                        id="nav-mobile"
                        style={
                          Object {
                            "width": "330px",
                          }
                        }
                      >
                        <li>
                          <Link
                            replace={false}
                            to="/home"
                          >
                            <a
                              href="/home"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-home"
                              >
                                 Home
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li>
                          <Link
                            className="dropdown-button"
                            data-activates="dropdown2"
                            replace={false}
                            to="!#"
                          >
                            <a
                              className="dropdown-button"
                              data-activates="dropdown2"
                              href="!#"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-user-circle-o"
                              >
                                 
                                User Account
                                 
                                <i
                                  aria-hidden="true"
                                  className="fa fa-angle-down"
                                />
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li>
                          <Link
                            onClick={[Function]}
                            replace={false}
                            to="!#"
                          >
                            <a
                              href="!#"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-sign-out"
                              >
                                 Logout
                              </i>
                            </a>
                          </Link>
                        </li>
                        <ul
                          className="dropdown-content"
                          id="dropdown2"
                          style={
                            Object {
                              "marginTop": "50px",
                            }
                          }
                        >
                          <li>
                            <Link
                              replace={false}
                              to="/profile"
                            >
                              <a
                                href="/profile"
                                onClick={[Function]}
                              >
                                <i
                                  aria-hidden="true"
                                  className="fa fa-user-circle"
                                >
                                   
                                   Profile
                                </i>
                              </a>
                            </Link>
                          </li>
                          <li
                            className="divider"
                          />
                          <li>
                            <Link
                              replace={false}
                              to="/addrecipe"
                            >
                              <a
                                href="/addrecipe"
                                onClick={[Function]}
                              >
                                <i
                                  aria-hidden="true"
                                  className="fa fa-plus-square"
                                >
                                   
                                   Add Recipe
                                </i>
                              </a>
                            </Link>
                          </li>
                          <li
                            className="divider"
                          />
                          <li>
                            <Link
                              replace={false}
                              to="/myrecipes"
                            >
                              <a
                                href="/myrecipes"
                                onClick={[Function]}
                              >
                                <i
                                  aria-hidden="true"
                                  className="fa fa-cutlery"
                                >
                                   
                                   My Recipes
                                </i>
                              </a>
                            </Link>
                          </li>
                          <li
                            className="divider"
                          />
                          <li>
                            <Link
                              replace={false}
                              to="/favourites"
                            >
                              <a
                                href="/favourites"
                                onClick={[Function]}
                              >
                                <i
                                  aria-hidden="true"
                                  className="fa fa-heart"
                                >
                                   
                                   Favourites
                                </i>
                              </a>
                            </Link>
                          </li>
                        </ul>
                      </ul>
                      <ul
                        className="side-nav"
                        id="mobile2"
                      >
                        <li>
                          <Link
                            replace={false}
                            to="/home"
                          >
                            <a
                              href="/home"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-home"
                              >
                                Home
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li
                          className="divider"
                        />
                        <li>
                          <Link
                            replace={false}
                            to="/profile"
                          >
                            <a
                              href="/profile"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-user-circle"
                              >
                                Profile
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li
                          className="divider"
                        />
                        <li>
                          <Link
                            replace={false}
                            to="/addrecipe"
                          >
                            <a
                              href="/addrecipe"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-plus-square"
                              >
                                Add Recipe
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li
                          className="divider"
                        />
                        <li>
                          <Link
                            replace={false}
                            to="/myrecipes"
                          >
                            <a
                              href="/myrecipes"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-cutlery"
                              >
                                My Recipes
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li
                          className="divider"
                        />
                        <li>
                          <Link
                            replace={false}
                            to="/favourites"
                          >
                            <a
                              href="/favourites"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-heart"
                              >
                                Favourites
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li
                          className="divider"
                        />
                        <li>
                          <Link
                            id="signOut"
                            onClick={[Function]}
                            replace={false}
                            to="!#"
                          >
                            <a
                              href="!#"
                              id="signOut"
                              onClick={[Function]}
                            >
                              <i
                                aria-hidden="true"
                                className="fa fa-sign-out"
                              >
                                 Logout
                              </i>
                            </a>
                          </Link>
                        </li>
                        <li
                          className="divider"
                        />
                      </ul>
                    </div>
                  </nav>
                </div>
              </Header>
            </Connect(Header)>
            <div
              className="container cont"
            >
              <div
                className="row"
              >
                <div
                  className="col s12  s2 m8 l8 offset-m2 offset-l2"
                >
                  <h2
                    style={
                      Object {
                        "textAlign": "center",
                      }
                    }
                  >
                    Sign Up
                  </h2>
                  <Connect(SignupForm)
                    history={
                      Object {
                        "push": [Function],
                      }
                    }
                  >
                    <SignupForm
                      error={Object {}}
                      history={
                        Object {
                          "push": [Function],
                        }
                      }
                      signup={[Function]}
                    >
                      <div
                        className="row"
                      >
                        <form
                          className="col s12 z-depth-2"
                          id="signup"
                          onSubmit={[Function]}
                          style={
                            Object {
                              "padding": "50px",
                            }
                          }
                        >
                          <div
                            className="row"
                          >
                            <div
                              className="input-field col s12 m6 l6"
                            >
                              <i
                                className="material-icons prefix"
                              >
                                account_circle
                              </i>
                              <input
                                className="validate"
                                id="fname"
                                name="firstName"
                                onChange={[Function]}
                                type="text"
                                value=""
                              />
                              <label
                                htmlFor="fname"
                              >
                                First Name
                              </label>
                            </div>
                            <div
                              className="input-field col s12 m6 l6"
                            >
                              <i
                                className="material-icons prefix"
                              >
                                account_circle
                              </i>
                              <input
                                className="validate"
                                id="lname"
                                name="lastName"
                                onChange={[Function]}
                                type="text"
                                value=""
                              />
                              <label
                                htmlFor="lname"
                              >
                                Last Name
                              </label>
                            </div>
                          </div>
                          <div
                            className="row"
                          >
                            <div
                              className="input-field col s12 m6 l6"
                            >
                              <i
                                className="material-icons prefix"
                              >
                                email
                              </i>
                              <input
                                className="validate"
                                id="email"
                                name="email"
                                onChange={[Function]}
                                type="text"
                                value=""
                              />
                              <label
                                htmlFor="email"
                              >
                                Email
                              </label>
                            </div>
                            <div
                              className="input-field col s12 m6 l6"
                            >
                              <i
                                className="material-icons prefix"
                              >
                                contact_phone
                              </i>
                              <input
                                className="validate"
                                id="phone"
                                name="phone"
                                onChange={[Function]}
                                type="text"
                                value=""
                              />
                              <label
                                htmlFor="phone"
                              >
                                Phone
                              </label>
                            </div>
                          </div>
                          <div
                            className="row"
                          >
                            <div
                              className="input-field col s12 m6 l6"
                            >
                              <i
                                className="material-icons prefix"
                              >
                                vpn_key
                              </i>
                              <input
                                className="validate"
                                id="password"
                                name="password"
                                onChange={[Function]}
                                type="password"
                                value=""
                              />
                              <label
                                htmlFor="password"
                              >
                                Password
                              </label>
                            </div>
                            <div
                              className="input-field col s12 m6 l6"
                            >
                              <i
                                className="material-icons prefix"
                              >
                                vpn_key
                              </i>
                              <input
                                className="validate"
                                id="confirmPassword"
                                name="confirmPassword"
                                onChange={[Function]}
                                type="password"
                                value=""
                              />
                              <label
                                htmlFor="confirmPassword"
                              >
                                Confirm Password
                              </label>
                            </div>
                          </div>
                          <div
                            className="row"
                          >
                            <Link
                              className="waves-effect waves-light btn"
                              id="signupBtn"
                              onClick={[Function]}
                              replace={false}
                              role="button"
                              tabIndex="0"
                              to="!#"
                            >
                              <a
                                className="waves-effect waves-light btn"
                                href="!#"
                                id="signupBtn"
                                onClick={[Function]}
                                role="button"
                                tabIndex="0"
                              >
                                Sign Up
                              </a>
                            </Link>
                          </div>
                          <div
                            className="row"
                          >
                            <p
                              className=""
                            >
                              Already have an account?
                               
                              <Link
                                replace={false}
                                to="/signin"
                              >
                                <a
                                  href="/signin"
                                  onClick={[Function]}
                                >
                                   Sign In
                                </a>
                              </Link>
                            </p>
                          </div>
                        </form>
                      </div>
                    </SignupForm>
                  </Connect(SignupForm)>
                </div>
              </div>
            </div>
            <Footer>
              <footer
                className="foot"
              >
                <p
                  className="foo"
                >
                  Copyright © 2017, More Recipes, All rights reserved
                </p>
              </footer>
            </Footer>
          </div>
        </SignupPage>
      </Connect(SignupPage)>
    </Router>
  </BrowserRouter>
</Provider>
`;
